### PR TITLE
Fix setting of resources for the CSI node driver

### LINKED ISSUE #2412: Adjust resources of csi-node-driver DaemonSet
## Current Behavior

It is not possible to specify the resources of the csi-node-driver DaemonSet using the Installation reference for the helm chart.

## Possible Solution

Add a `csiNodeDriverDaemonSet` option to the Installation reference that overrides the resource field for the mentioned DaemonSet

## Context

All other resources created by the tigera operator have an option to adjust the resources except the csi-node-driver

### PR DESCRIPTION
**Pull request by yeahservice**
Fixes #2412 

* Change installation CRD to allow the CSI node driver container names
* Change validation logic for the CSI node driver to allow resources

## For PR author

- [x] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by tmjd** [comment]
I tried fixing a conflict and apparently fixed it the wrong way, @yeahservice could you pull down the update and fix the

**Comment by tmjd** [comment]
Sorry, I didn't mean to close this. I was clicking and hit the close button.

**Comment by yeahservice** [comment]
> I tried fixing a conflict and apparently fixed it the wrong way, @yeahservice could you pull down the update and fix the

Looks fine to me, am I missing something?

**Comment by tmjd** [comment]
> Looks fine to me, am I missing something?

Sorry @yeahservice I think I had tried something it didn't work, I added that comment, and then I realized my problem and made the update that was needed but forgot to go back and remove/edit that comment.

File: api/v1/csi_node_driver.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 // CSINodeDriverDaemonSetContainer is a csi-node-driver DaemonSet container.
 type CSINodeDriverDaemonSetContainer struct {
 	// Name is an enum which identifies the csi-node-driver DaemonSet container by name.
-	// +kubebuilder:validation:Enum=csi-node-driver
+	// +kubebuilder:validation:Enum=calico-csi;csi-node-driver-registrar
```
[the remark this answers is not in the document]
Comment (yeahservice) [reply api/v1/csi_node_driver.go:27]: Good point, I will test the migration path as well and check if it is needed.

[the remark this answers is not in the document]
Comment (caseydavenport) [reply api/v1/csi_node_driver.go:27]: Yeah strictly speaking we should leave the old value, at least for several releases with a formal deprecation and then we can remove it.

**Comment by gnuletik** [comment]
Thanks for this PR! I'm also trying to set the resources on this component.
Is there anything blocking this?
Thanks!

**Comment by caseydavenport** [comment]
I think we're still waiting on a change to add back the `csi-node-driver` value, as well as a rebase to fix merge conflicts.

**Comment by caseydavenport** [comment]
Fixes https://github.com/tigera/operator/pull/2768

**Comment by caseydavenport** [comment]
@yeahservice are you still planning to finish this one off? Looks like it needs a rebase to fix some conflicts, and there's one small change required.

**Comment by caseydavenport** [comment]
Unfortunately going to close this one due to inactivity and conflicts. Feel free to re-raise another PR, though.

**Pull request closed without merging**